static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -189,12 +189,15 @@ function useReplayData({replaySlug, orgSlug}: Options): Result {
     setState(INITIAL_STATE);
 
     try {
-      const [record, attachments] = await Promise.all([
-        fetchReplay(),
+      const [[replayRecord, errors], attachments] = await Promise.all([
+        (async (): Promise<[ReplayRecord, any]> => {
+          const fetchedRecord = await fetchReplay();
+          const mappedRecord = mapResponseToReplayRecord(fetchedRecord);
+          const fetchedErrors = await fetchErrors(mappedRecord);
+          return [mappedRecord, fetchedErrors];
+        })(),
```

Comment:
Imo this is a bit hard to read with nested array destructuring and the iife... what about something like:

```javascript
const replayPromise = fetchReplay();
const attachmentsPromise = fetchAllRRwebEvents();
const fetchedRecord = await replayPromise;
const mappedRecord = mapResponseToReplayRecord(fetchedRecord);
const [errors, attachments] = await Promise.all([fetchErrors(mappedRecord), attachmentsPromise])
```

Reply:
it's not my preference to split up promise exec from the await call, i think it makes it easy to have different things mix together, for example i'm not a fan of:
```
const replayPromise = fetchReplay();
... anything can happen! the two parts are not connected enough for me ...
const fetchedRecord = await replayPromise;
```


It'd give the IIFE a name though, then we could do two destructuring's... along the lines of:
```typescript
const fetchReplayAndErrors = async (): Promise<[ReplayRecord, any]> => {
  const fetchedRecord = await fetchReplay();
  const mappedRecord = mapResponseToReplayRecord(fetchedRecord);
  const fetchedErrors = await fetchErrors(mappedRecord);
  return [mappedRecord, fetchedErrors];
});

const [replayAndErrors, attachments] = await Promise.all([fetchReplayAndErrors(), fetchAllRRwebEvents()]);
const [replay, errors] = replayAndErrors;
```